components/eam/src/physics/cam/shoc_intr.F90

```diff
@@ -81,11 +83,18 @@ module shoc_intr
       shoc_ice_sh = 50.e-6  
          
   logical      :: lq(pcnst)
-  logical      :: lq2(pcnst)
+
+  !lq_dry_wet_cnvr is true for all the water constituents used  by SHOC.
+  !These water constituents will participate in dry/wet mmr conversion
+  logical      :: lq_dry_wet_cnvr(pcnst)
  
   logical            :: history_budget
   integer            :: history_budget_histfile_num  
   logical            :: micro_do_icesupersat
+
+  !Store names of the water constituents used by SHOC
+  character(len=6), parameter :: dry_wet_scalars(9) = ['Q     ','CLDLIQ', &
```

Comment:
Can you confirm that SHOC uses all of these tracers?  Including RAINQM, CLDRIM, NUMRAI, BVRIM?  If so then we have a little more work cut out for us because SHOC should be agnostic to the microphysics scheme and CLDRIM and BVRIM are specific to P3 only.  MG2 has a SNOW constituent instead.  So if we actually need these here then we need a conditional to check the `microp_scheme` and act accordingly.  Otherwise the conditional loop down below will fail with MG2 to capture SNOW.

Reply:
My original idea was to have this array and add all the "possible" water constituents (which includes water constituents from all the microphysics schemes such as MG) in this array. Later in the code, we can check if a water constituent in the Q array exists in this list or not. If it exists, it can participate in the wet/dry conversion. 

From your comment above, I agree with you that it may be too restrictive. Another idea is to let all constituents in the Q array participate in the wet/dry conversion except for the TKE. Do you think that is a viable solution? That will let SHOC run with any microphysics scheme.

Reply:
Perhaps we can hard-code P3 stuff for scream. But for EAMxx, we might want to have a "water_constituent" group, so that SHOC can be used without assuming anything on which constituents are actually in the model.

Reply:
Yeah, it will be better to have such a tag or group for the C++ codes. Currently C++ codes just exclude TKE from the dry/wet conversion.

We can use similar approach (of excluding TKE) in the F90 code.

Reply:
Yeah, I think if we go for the exclusion approach it would be safer.  I think you can grab a string array of the constituent names and then search through them and exclude TKE.  That way if we have SNOW (MG2) or RIME (P3) the SHOC process won't care.  It will just grab all constituents except TKE.  Does that make sense?